pkg/splunk/client/remotedataclient.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2018-2022 Splunk Inc. All rights reserved.
+
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// RemoteDataClientsMap is a map of remote storage provider name to
+// their respective initialization procedures
+// Currently supported remote data clients are:
+// aws
+// minio
+// azure
+// in future we may have
+// googlestorage
+var RemoteDataClientsMap = make(map[string]GetRemoteDataClientWrapper)
+
+// RemoteObject struct contains contents returned as part of remote data client response
+// for example, it can be a blob in azure or s3 object in aws/minio
+type RemoteObject struct {
+	Etag         *string
+	Key          *string
+	LastModified *time.Time
+	Size         *int64
+	StorageClass *string
+}
+
+// RemoteDataListRequest struct contains inputs specifying storage account
+//
+type RemoteDataListRequest struct {
+}
+
+// RemoteDataListResponse struct contains list of RemoteObject objects
+type RemoteDataListResponse struct {
+	Objects []*RemoteObject
+}
+
+// RemoteDataDownloadRequest struct specifies the remote data file path,
+// local file where the downloaded data should be written as well as
+// the authontication data if available
+type RemoteDataDownloadRequest struct {
+	LocalFile  string // file name where the remote data will be written
+	RemoteFile string // file name with path relative to the bucket
+	Etag       string // unique tag of the object
+}
+
+// RemoteDataClient is an interface to provide
+// listing and downloading of apps from remote data storage
+//
+type RemoteDataClient interface {
+
+	// Get the list of Apps
```

Comment:
nit: I think, generally, we call it as App package, instead of App. Same with other places.

Reply:
changed